diff --git a/config.mk b/config.mk
index 95328ba..f2c61d7 100644
--- a/config.mk
+++ b/config.mk
@@ -4,7 +4,7 @@ VERSION = 1.5
 # Customize below to fit your system
 
 # paths
-PREFIX = /usr/local
+PREFIX = /home/jack
 MANPREFIX = ${PREFIX}/share/man
 
 X11INC = /usr/X11R6/include
diff --git a/slock.c b/slock.c
index 0ba1098..3b06ca8 100644
--- a/slock.c
+++ b/slock.c
@@ -221,6 +221,14 @@ readpw(Display *dpy, struct xrandr *rr, struct lock **locks, int nscreens,
 				}
 				XSync(dpy, False);
 
+				// If a password has been entered, set RHOST
+				// This will cause pam to skip authentication methods like fingerprints
+				// An alternative would be to have two different pam service files
+				// or to use pam-fprint-grosshack, but this way is nice and simple
+				if (len > 0) {
+					pam_set_item(pamh, PAM_RHOST, "_dummy");
+				}
+
 				if (retval == PAM_SUCCESS)
 					retval = pam_authenticate(pamh, 0);
 				if (retval == PAM_SUCCESS)
